src/autoscaler/pruner/pruner.go

```diff
@@ -1,56 +1,48 @@
 package pruner
 
 import (
+	"os"
 	"time"
 
 	"code.cloudfoundry.org/clock"
 	"code.cloudfoundry.org/lager"
 )
 
-type Pruner interface {
-	PruneOldData()
+type DbPruner interface {
+	Prune()
 }
 
 type DbPrunerRunner struct {
-	pruner   Pruner
+	dbPruner DbPruner
 	name     string
 	interval time.Duration
 	clock    clock.Clock
-	doneChan chan bool
 	logger   lager.Logger
 }
 
-func NewDbPrunerRunner(pruner Pruner, name string, interval time.Duration, clock clock.Clock, logger lager.Logger) *DbPrunerRunner {
+func NewDbPrunerRunner(dbPruner DbPruner, name string, interval time.Duration, clock clock.Clock, logger lager.Logger) *DbPrunerRunner {
 	return &DbPrunerRunner{
-		pruner:   pruner,
+		dbPruner: dbPruner,
 		name:     name,
 		interval: interval,
 		clock:    clock,
 		logger:   logger,
```

Comment:
Create the logger with the name as the session.
Clean up all logger usage since the name will be there.